### PR TITLE
1.add config content check, 2.replace ZMQ with boost mq

### PR DESCRIPTION
**Pull request by ericyx**

### CODE REVIEW

File: src/common/Config.cpp [hunk 1]
```diff
@@ -57,6 +57,15 @@ Config* get_config(int* argc, char*** argv)
...
       config->jnldevs,
   };
+  if (cfg.id.empty() ||
+    cfg.role.empty() ||
+    cfg.master.empty() ||
+    cfg.zookeeper.empty() ||
+    cfg.restport == 0 ||
+    cfg.fileport == 0 ||
+    cfg.fileroot.empty()){
+      throw (errno);
```
Comment (zyluo) [inline src/common/Config.cpp]: Where is errno defined?

File: src/scheduler/EventLoop.cpp [hunk 2]
```diff
@@ -33,19 +33,21 @@ EventLoop::~EventLoop()
...
   try{
     while (loopTag) {
-      zmq::message_t message;
-      string event_data;
-      receiver.recv(&message);
-      istringstream iss(static_cast<char*>(message.data()));
-      iss >> event_data;
-      LOG(INFO) << "EventLoop.recvData got flexUp data";
-      processData(event_data);
+      char event_data[256];
```
Comment (zyluo) [inline src/scheduler/EventLoop.cpp]: let's think about whether we need a bigger size than 256.

**Comment by zyluo** [comment]
I remember there is also a zmqcontext in /tests/JsonTest.cpp

File: src/common/Config.cpp (hunk 1 above)
Comment (ericyx) [inline src/common/Config.cpp]: so I need find a better way to throws exception and define it by us, for example out "config content error"

File: src/scheduler/EventLoop.cpp (hunk 2 above)
Comment (ericyx) [inline src/scheduler/EventLoop.cpp]: ok,

**Pull request closed without merging**